compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -691,6 +691,25 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         {
                             return;
                         }
+                        if let ObligationCauseCode::FunctionArgumentObligation {
```

Comment:
This doesn't deduplicate usages of unsized bindings in other positions that also require the expression to be sized, such as struct constructors, assignment, `as` coercions, returns, etc right?

Reply:
Correct.